pkg/northbound/gnmi/subscribe.go

```diff
@@ -129,38 +129,35 @@ func broadcastNotification() {
 		log.Println("Error while subscribing to updates", err)
 	}
 	for update := range changes {
-		//TODO needs to be filtered for appropriate paths in the change
-		// currently broadcasting to everybody
-		for _, ch := range PathToChannels {
-			values := update.Values()
-			target := update.Subject()
-			changeID := (*values)[events.ChangeID]
-			changeInternal := mgr.ChangeStore.Store[changeID]
-			err := sendUpdate(target, changeInternal, ch)
-			if err != nil {
-				log.Println("Error while parsing path ", err)
+		target, changeInternal := getChange(update, mgr)
+		//For every channel we cycle over the paths of the config change and if somebody is subscribed to it we send out
+		for path, ch := range PathToChannels {
+			for _, changeValue := range changeInternal.Config {
+				//FIXME this might prove expensive, find better way to store path and targer in channels map
```

Comment:
I don't mind this - you're going between 2 different paradigms for dealing with paths